covasim/datasets/data_loader.py

```diff
@@ -0,0 +1,127 @@
+import requests
+import os
+import json
+import numpy as np
+dirname = os.path.dirname(__file__)
+mapper_file_path = os.path.join(dirname,"sources/lookup.json")
+
+##
+# DataLoader: Abstract class that formats the common
+# interface for uploading an external data source and
+# translating to the desired format.
+#
+class DataLoader:
+    def __init__(self):
+        pass
+
+    ##
+    # update_data: pull the external data local calls
+    # translate with the data from the external source
+    # and saves it locally. Only needs to be called
+    # once ever, unless the external source changes
+    #
+    def update_data(self):
+        data_json = self.fetch_data()
+        self.delete_data()
+        translated = self.translate(data_json)
+        with open(self.file_path(), "a") as outfile:
+            json.dump(translated, outfile)
+
+    ##
+    # fetch_data: gets eternal data and casts
+    # it to json. May need pandas or somethig else
+    # eventually.
+    #
+    def fetch_data(self):
+        r = requests.get(self.URL)
+        return r.json()
+
+    ##
+    # delete_data: delete the locally stored trnaslated file.
+    #
+    def delete_data(self):
+        if os.path.exists(self.file_path()):
+            os.remove(self.file_path())
+
+    ##
+    # file_path: path to the locally stored translated data
+    #
+    def file_path(self):
+        return os.path.join(dirname,"sources/population/{0}.json".format(self.FILENAME))
+
+    ##
+    # Load the locally stored translated data into cache.
+    #
+    def load_data(self):
+        if hasattr(self, 'data'):
+            return
+        with open(self.file_path()) as datafile:
+            strdata = json.load(datafile)
+        self.data = strdata
+    ##
+    # translate: a function that will take an external data source
+    # and translate it to the desired format.
+    #
+    def translate(self, json):
+        raise NotImplementedError
+
+class NeherLabPop(DataLoader):
```

Comment:
This class is cool -- I do wonder though if we can start with this just as a simple script instead, and skip the DataLoader class

Reply:
I could just merge the DataLoader class with the NeherLabPop class... The actual script we run wouldn't change any.

Reply:
sounds good